### PR TITLE
Support editing existing questions

### PR DESCRIPTION
**Pull request by hmhsu**
### Description
- We update the existing question without changing its version. Path and id cannot be changed.
- handle any exceptions by surfacing them to users with flash
- disallow blank path

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
[Fixes] #103

### CODE REVIEW

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java [hunk 1]
```diff
@@ -54,14 +56,21 @@ public QuestionController(
...
+                if (!success) {
+                  exception =
+                      String.format(
+                          "create failed: this is most likely you specify an invalid path %s",
+                          definition.getPath());
+                }
               } catch (UnsupportedQuestionTypeException e) {
-                // I'm not sure why this would happen here, so we'll just log and redirect.
+                exception = e.toString();
                 LOG.info(e.toString());
```
Comment (AWBaumann) [inline universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:71]: No sense in calling `e.toString()` twice. Just use `LOG.info(exception)`

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java [hunk 2]
```diff
@@ -89,38 +98,47 @@ public Result newOne(Request request) {
...
     Form<QuestionForm> form = formFactory.form(QuestionForm.class);
     QuestionForm questionForm = form.bindFromRequest(request).get();
+    String exception = "";
     try {
       QuestionDefinition definition = questionForm.getBuilder().setId(id).setVersion(1L).build();
       service.update(definition);
     } catch (UnsupportedQuestionTypeException e) {
-      // I'm not sure why this would happen here, so we'll just log and redirect.
+      exception = e.toString();
       LOG.info(e.toString());
```
Comment (AWBaumann) [inline universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:129]: Same as above. `Log.info(exception);`

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java [hunk 3]
```diff
@@ -89,38 +98,47 @@ public Result newOne(Request request) {
...
-    } catch (UnsupportedOperationException e) {
-      // This is expected for now until we implement update on QuestionService.
+    } catch (InvalidUpdateException e) {
+      exception = e.toString();
+      LOG.info(e.toString());
```
Comment (AWBaumann) [inline universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:132]: `Log.info(exception);`

**Review by AWBaumann — APPROVED** [review]
Nice work! LGTM!

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java (hunk 2 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java]: Done.

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java (hunk 3 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java]: Done.

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java (hunk 1 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java]: Done.

**Pull request merged**